### ISSUE TITLE
Access weights learned by NNConv & SplineConv

### ISSUE THREAD

**Issue by adhikarirsr** [pyg-team/pytorch_geometric#1461]
In the case of `GCNConv`, I can access the values of weights learned by using `modelgcn.conv1.weight.data`.However, for `NNConv` and `GMMConv`, I can't do so because `NNConv` does not have an attribute `weight`. Is there a way to access `weight` for NNConv? I want to investigate what kind of filter is learned by `NNConv` and `GMMConv`.

Also, how do I go about plotting the filter for `SplineConv`?

For a network like this
```
class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = SplineConv(1, 64, dim=2, kernel_size=3)
        self.conv2 = SplineConv(64, 64, dim=2, kernel_size=3)
        self.conv3 = SplineConv(64, 64, dim=2, kernel_size=3)
        self.conv4 = SplineConv(64, 1, dim=2, kernel_size=3)
```
I can access weights by this command: `wl1= modelscn.conv1.weight` and `wl1.shape` give me this `torch.Size([9, 1, 64])`.  I guess that there is 9 because the kernel size is  3 (which is 3X3) and 64 is because there are 64 filters of size 3x3. How do I go about plotting them? Which filter value represent which direction?

**Comment by rusty1s** [comment]
The filter in `NNConv` is an MLP, and you can access it via `self.conv.nn` (containing the module you passed to the `nn` argument of `NNConv`). You can visualize the learned filters by executing `self.conv.nn` for various edge features, *e.g.*:
```python
edge_attr = torch.randn(1000, 3)
weight = self.conv.nn(edge_attr).view(-1, self.in_channels, self.out_channels)
```

**Comment by psimeson123** [comment]
Thanks @rusty1s  for replying.

Here's what I tried to do. Please let me know if my understanding is sensible or not.

Here's a network, I am playing with:
```
class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        nn1 = nn.Sequential(nn.Linear(2, 25), nn.ReLU(), nn.Linear(25, 16))
        self.conv1 = NNConv(1, 16, nn1, aggr='add')
        self.bn1 = BatchNorm(16)

        nn2 = nn.Sequential(nn.Linear(2, 25), nn.ReLU(), nn.Linear(25, 256))
        self.conv2 = NNConv(16, 16, nn2, aggr='add')
        self.bn2 = BatchNorm(16)
        
        nn3 = nn.Sequential(nn.Linear(2, 25), nn.ReLU(), nn.Linear(25, 256))
        self.conv3 = NNConv(16, 16, nn3, aggr='add')
        self.bn3 = BatchNorm(16)

        nn4 = nn.Sequential(nn.Linear(2, 25), nn.ReLU(), nn.Linear(25, 16))
        self.conv4 = NNConv(16, 1, nn4, aggr='add')
```
1. For conv1, I have a neural network with two linear layers. In this case, there are  16 filters of size 5*5 (because of ` nn.Linear(25, 16)` specification) right?
2. This is what I did to plot the 1st filter learned by the neural net: `plt.imshow(modelnn.conv1.nn[2].weight[0].reshape(5,5).detach().numpy())`
Does this make sense?

**Comment by rusty1s** [comment]
1. You are right that there are 16 output filters for the first convolution, however, the `25` does not correspond to the filter size. The filter size is the 1-hop neighborhood around each node, while the `25` just represents the hidden dimensionality of your multi-layer filter function. Note that this hidden layer is shared **across all filters**.
2. I don't think plotting the hidden embedding is very useful. Instead, you want to plot the outcome of the neural network `nn` for different edge features, *e.g.*:
```python
pos = torch.randn(1000, 2)  # The number of points we want to sample from the filter
out = modelnn.conv1.nn(pos)
```
This will get you, for each point in `pos`, 16 output values which correspond to the 16 filters you learn.
